File: src/main/java/me/elsiff/morefish/condition/TimeCondition.java
package me.elsiff.morefish.condition;

import org.bukkit.entity.Player;

public class TimeCondition implements Condition {
    private final String time;

    public TimeCondition(String time) {
        this.time = time;
    }

    @Override
    public boolean isSatisfying(Player player) {
        long tick = player.getWorld().getTime();
        switch (time) {
            case "day":
                return (1000 <= tick && tick < 13000);
            case "night":
                return (13000 <= tick || tick < 1000);
            default:
                return false;
        }
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/TimeCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import org.bukkit.entity.Item
import org.bukkit.entity.Player

class TimeCondition(
    private val state: TimeState
) : FishCondition {
    override fun check(
        caught: Item,
        fisher: Player,
        competition: FishingCompetition
    ): Boolean {
        return TimeState.fromTime(caught.world.time) == state
    }

    enum class TimeState(
        vararg val range: IntRange
    ) {
        DAY(1000 until 13000),
        NIGHT(0 until 1000, 13000 until 24000);

        companion object {
            fun fromTime(worldTime: Long): TimeState? {
                return values().find { state ->
                    state.range.any { it.contains(worldTime) }
                }
            }
        }
    }
}
